### ISSUE TITLE
Feature: publish arm64 builds to Docker Hub

### ISSUE THREAD

**Issue by hazcod** [cockroachdb/cockroach#60778]
**Is your feature request related to a problem? Please describe.**
Running the amd64 build on Docker for Mac on Apple Silicon is very slow.

**Describe the solution you'd like**
Push native arm64 builds to Docker Hub as wel.

https://hub.docker.com/r/cockroachdb/cockroach/tags?page=1&ordering=last_updated

Epic CRDB-1463

Jira issue: CRDB-3143

**Comment by dt** [comment]
This will need to wait at least until we upgrade to go 1.16 -- some after 21.1 is released -- for Darwin arm64 support.

**Comment by anr** [comment]
Hello @dt, I believe just releasing a Linux arm64 build for docker would help Apple Silicon users - and other docker users on arm64.

**Comment by FireMasterK** [comment]
According to https://github.com/cockroachdb/cockroach/issues/20433#issuecomment-634205077, building arm64 images is already possible, so why not have officially built images?

**Comment by lululombard** [comment]
Up for this, I'd like to deploy a K8s Raspberry Pi cluster in a few months, so I'd kinda need this

**Comment by akkie** [comment]
Any news here? I think more and more people switching to ARM based Apple machines these days. So I think it's crucial to have ARM support out of the box for local development. The workaround is to use PostgreSQL with the subset that is supported from CockroachDB.

**Comment by hazcod** [comment]
Should also be not that much work due to the nature of Go and I think no C dependencies?

**Comment by rickystewart** [comment]
Unfortunately we do have C dependencies.

Here's an update I wrote in another issue a few weeks back: https://github.com/cockroachdb/cockroach/issues/58676#issuecomment-1049130427

Last week I took a few days to more fully explore ARM support and we now have a working cross-toolchain for M1 Macs. There are additional processes and infrastructure we need to get in place to be able to publish ARM64 cockroach images. I hope that we can have that done by the 22.2 release but it's not certain -- planning is currently underway, though.

**Comment by jlinder** [comment]
Hi, here's a short update to confirm that we plan to include _experimental_ Linux ARM64 binaries and docker images, with the 22.2 release series coming this fall.

**Comment by jbg** [comment]
Just to make sure you don't focus too much on M1 Macs (although native support there would be great too): I just started work on deploying a pilot of CockroachDB at work and hit a wall when I realised you don't publish linux/arm64 images. Our Kubernetes clusters at work have been arm64-only (Graviton) since more than a year now, for better performance and lower price. I'm sure we're not the only ones. This is actually the first time in recent memory that we tried to deploy something and didn't find arm64 images already published.

We'll just build from source for now, but I wanted to point out that this isn't "just" a developer experience thing for running it on M1 Macs, there are plenty of production arm64 deployments around.
